pre_commit_hooks/check_permitted_path_characters.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+import argparse
+import string
+from typing import Sequence
+
+DEFAULT_ALLOWLIST = string.ascii_letters + string.digits + '-_./'
```

Comment:
`os.path.sep` or are paths unix-normalized by the time they arrive at the hook scripts?